src/debug-artifacts.ts

```
@@ -233,23 +246,16 @@ export async function uploadDebugArtifacts(
     }
   }
 
-  try {
-    await artifact.create().uploadArtifact(
-      sanitizeArtifactName(`${artifactName}${suffix}`),
-      toUpload.map((file) => path.normalize(file)),
-      path.normalize(rootDir),
-      {
-        continueOnError: true,
-        // ensure we don't keep the debug artifacts around for too long since they can be large.
-        retentionDays: 7,
-      },
-    );
-  } catch (e) {
```

Since we removed this `catch`, is it correct that now we're not suppressing the failure and instead surface it, failing the run?

Why did we change this behaviour? The change makes sense to me, but it looks potentially breaking for some flows.

---

We're now catching the exception in the callers.  The main reason for changing it is to provide a more descriptive error message, since the caller knows what _type_ of debugging artifact we were trying to upload e.g. combined SARIF file, or the regular debug artifacts.